`gcp.project.kms` resource will run into an error on keys with the status not available

**Describe the bug**
`gcp.project.kms {keyrings {name cryptokeys {name rotationPeriod <= 90*time.day }}}` will run into an error on keys with the status not available:

![image](https://user-images.githubusercontent.com/112621871/213766346-5bb53fbf-e016-45ab-8391-0bffd89dfabb.png)

## Error message:

```
cnquery> gcp.project.kms {keyrings {name cryptokeys {name rotationPeriod <= 90*time.day }}}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x69c1db9]

goroutine 58 [running]:
go.mondoo.com/cnquery/resources/packs/gcp.cryptoKeyVersionToMql(0xc001e6f7c0?, 0x0)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/resources/packs/gcp/kms.go:384 +0x39
go.mondoo.com/cnquery/resources/packs/gcp.(*mqlGcpProjectKmsServiceKeyring).GetCryptokeys(0xc0014bc030)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/resources/packs/gcp/kms.go:247 +0x471
go.mondoo.com/cnquery/resources/packs/gcp.(*mqlGcpProjectKmsServiceKeyring).ComputeCryptokeys(0xc0014bc030)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/resources/packs/gcp/gcp.lr.go:27883 +0x5e
go.mondoo.com/cnquery/resources/packs/gcp.(*mqlGcpProjectKmsServiceKeyring).MqlCompute(0xc00049ac80?, {0xc000da6b50, 0xa})
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/resources/packs/gcp/gcp.lr.go:27871 +0x177
go.mondoo.com/cnquery/resources.(*Runtime).WatchAndUpdate(0xc00049ad40, {0x9b20540, 0xc0014bc030}, {0xc000da6b50, 0xa}, {0xc001e2a580, 0x32}, 0xc00164e330)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/resources/runtime.go:345 +0x39b
go.mondoo.com/cnquery/llx.runResourceFunction(0xc000fe2930, 0xc00164e090, 0xc000a634f0, 0x300000003)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/builtin.go:712 +0x23e
go.mondoo.com/cnquery/llx.(*blockExecutor).runBoundFunction(0xc000fe2930, 0xc00164e090, 0xc000a634f0, 0xc00049d0b0?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/builtin.go:795 +0xf2
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc000fe2930, 0xc000a634f0, 0x0?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:764 +0xde
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc000fe2930, 0xc00164e1e0?, 0x300000003?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc000cf4670?, 0xc?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc000fe2930, 0x8949bb9?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).run(0xc000fe2930)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:351 +0x2d7
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunctionBlock(0xc000330d90, {0xc001e261f0, 0x1, 0xc00164e0c0?}, 0x300000000, 0x0?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:582 +0x12f
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunctionBlocks(0xc000330d90, {0xc0003f0db0, 0x1, 0xc0013ea650?}, 0x496805?, 0xc00166e480?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:553 +0x11b
go.mondoo.com/cnquery/llx.arrayBlockListV2(0xc000330d90, 0xc00164e000, 0xc000a638b0, 0x200000003)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/builtin_array.go:143 +0x5c8
go.mondoo.com/cnquery/llx.(*blockExecutor).runBoundFunction(0xc000330d90, 0xc00164e000, 0xc000a638b0, 0x0?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/builtin.go:782 +0x119
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc000330d90, 0xc000a638b0, 0x400?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:764 +0xde
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc000330d90, 0xc00126f830?, 0x200000003?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc000c70930?, 0xe?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc000330d90, 0xc00126f7d0?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).triggerChain(0xc000330d90, 0xc00126f7d0?, 0xc00164e000)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:910 +0x399
go.mondoo.com/cnquery/llx.runResourceFunction.func1({0x76126a0?, 0xc0003f0d98}, {0x0?, 0x0})
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/builtin.go:731 +0x245
go.mondoo.com/cnquery/resources.(*Runtime).WatchAndUpdate.func1()
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/resources/runtime.go:315 +0x258
go.mondoo.com/cnquery/resources.(*Runtime).WatchAndUpdate(0xc00049ad40, {0x9b20500, 0xc000967408}, {0xc000da6b20, 0x8}, {0xc000649440, 0x31}, 0xc00126fbf0)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/resources/runtime.go:354 +0x3f4
go.mondoo.com/cnquery/llx.runResourceFunction(0xc000330d90, 0xc00126f050, 0xc000a63400, 0x200000002)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/builtin.go:712 +0x23e
go.mondoo.com/cnquery/llx.(*blockExecutor).runBoundFunction(0xc000330d90, 0xc00126f050, 0xc000a63400, 0xc0008e6660?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/builtin.go:795 +0xf2
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc000330d90, 0xc000a63400, 0x30?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:764 +0xde
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc000330d90, 0xc00126f830?, 0x200000002?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc000c70930?, 0xc?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc000330d90, 0x8949bb9?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).run(0xc000330d90)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:351 +0x2d7
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunctionBlock(0xc00106f500, {0xc000967428, 0x1, 0xc00126f440?}, 0x200000000, 0x0?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:582 +0x12f
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunctionBlocks(0xc00106f500, {0xc0013eb190, 0x1, 0xc000ab98c0?}, 0x893e672?, 0x2?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:553 +0x11b
go.mondoo.com/cnquery/llx.(*blockExecutor).runBlock(0xc00106f500, 0xc00126f050, 0x8d95ba8?, {0xc0009da218, 0x0, 0x0?}, 0x100000003)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:617 +0x40d
go.mondoo.com/cnquery/llx.init.0.func1(0xc000afbbd8?, 0x17?, 0x893e672?, 0x2?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/builtin.go:646 +0x52
go.mondoo.com/cnquery/llx.(*blockExecutor).runBoundFunction(0xc00106f500, 0xc00126f050, 0xc000a63900, 0xc000c708a0?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/builtin.go:782 +0x119
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc00106f500, 0xc000a63900, 0xc000ca7878?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:764 +0xde
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc00106f500, 0xc00106b950?, 0x100000003?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc000cc0a80?, 0x771c1c0?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc00106f500, 0xc00106b920?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).triggerChain(0xc00106f500, 0xc00106b920?, 0xc00126f050)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:910 +0x399
go.mondoo.com/cnquery/llx.runResourceFunction.func1({0x8770220?, 0xc000967408}, {0x0?, 0x0})
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/builtin.go:731 +0x245
go.mondoo.com/cnquery/resources.(*Runtime).WatchAndUpdate.func1()
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/resources/runtime.go:315 +0x258
go.mondoo.com/cnquery/resources.(*Runtime).WatchAndUpdate(0xc00049ad40, {0x9b20100, 0xc0009da250}, {0xc000da6b07, 0x3}, {0xc000649400, 0x31}, 0xc00126e810)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/resources/runtime.go:354 +0x3f4
go.mondoo.com/cnquery/llx.runResourceFunction(0xc00106f500, 0xc00124cba0, 0xc000a63360, 0x100000002)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/builtin.go:712 +0x23e
go.mondoo.com/cnquery/llx.(*blockExecutor).runBoundFunction(0xc00106f500, 0xc00124cba0, 0xc000a63360, 0x2?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/builtin.go:795 +0xf2
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc00106f500, 0xc000a63360, 0x203000?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:764 +0xde
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc00106f500, 0xc00106b950?, 0x100000002?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc000cc0a80?, 0xc00106b920?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc00106f500, 0x8949bb9?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).run(0xc00106f500)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:351 +0x2d7
go.mondoo.com/cnquery/llx.(*MQLExecutorV2).Run(0xc00102a5a0?)
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/llx/llx.go:283 +0x59
go.mondoo.com/cnquery/mql/internal.(*executionManager).executeCodeBundle(0xc000a63950, 0xc000fa60a0, 0x0?, {0x0, 0x0})
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/mql/internal/execution_manager.go:167 +0x35d
go.mondoo.com/cnquery/mql/internal.(*executionManager).Start.func1()
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/mql/internal/execution_manager.go:85 +0x1cb
created by go.mondoo.com/cnquery/mql/internal.(*executionManager).Start
	/home/manuel/go/src/go.mondoo.io/mondoo/cnquery/mql/internal/execution_manager.go:54 +0x6a
(base) manuel@manuel-ThinkPad-X1-Carbon-Gen-9

```

Fixed/workaround by regenerating/rotating the faulty keys
![image](https://user-images.githubusercontent.com/112621871/213771881-6876c212-9788-4ffb-8e54-4b03f71401b1.png)

## Expected behaviour
Gracefully deal with the status error

## To Reproduce:
Fail at generating a key in the expected format:
![image](https://user-images.githubusercontent.com/112621871/213772550-285188b7-ec26-44fd-9fcb-061bb2b35d21.png)
The uploaded keyfile in this example is not generated with the pubkey material downloaded from the job.
This wil leave the key status as `not available`:
![image](https://user-images.githubusercontent.com/112621871/213772821-5583bfeb-ac7c-4129-a24b-2d759d23ff28.png)

--- a/resources/packs/gcp/kms.go
+++ b/resources/packs/gcp/kms.go
@@ -244,9 +244,12 @@ func (g *mqlGcpProjectKmsServiceKeyring) GetCryptokeys() ([]interface{}, error) 
 			return nil, err
 		}
 
-		mqlPrimary, err := cryptoKeyVersionToMql(g.MotorRuntime, k.Primary)
-		if err != nil {
-			return nil, err
+		var mqlPrimary interface{}
+		if k.Primary != nil {
+			mqlPrimary, err = cryptoKeyVersionToMql(g.MotorRuntime, k.Primary)
+			if err != nil {
+				return nil, err
+			}
 		}
 
 		var versionTemplate map[string]interface{}

